components/CredentialSwitch.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <component
+    :is="selection"
+    v-bind="{...$props, ...$attrs}">
+    <template
+      v-for="(_, slot) in $slots"
+      #[slot]="scope">
+      <slot
+        :name="slot"
+        v-bind="scope || {}" />
+    </template>
+  </component>
+</template>
+
+<script setup>
+import {computed, defineProps, unref} from 'vue';
+import {config} from '@bedrock/web';
+import CredentialBase from './CredentialBase.vue';
+import CredentialDetail from './CredentialDetail.vue';
+import {merge} from 'lodash-es';
```

Comment:
We've tried really hard to remove `lodash` from everything because of its various security maintenance issues. Is there a simpler local solution that can be used here -- or something else we could use like `extend` from `@bedrock/web`?